framework/test/unit/modules/interoperability/mainchain/endpoint.spec.ts

```diff
@@ -60,61 +65,82 @@ describe('MainchainInteroperabilityEndpoint', () => {
 				offchainStoresMock as any,
 			);
 			interopMod.stores.register(RegisteredNamesStore, registeredNamesStore as never);
+			jest.spyOn(registeredNamesStore, 'has').mockResolvedValue(true);
 		});
 
 		it('should throw error if name is not a string', async () => {
-			// Arrange
 			const context = createTransientModuleEndpointContext({
 				params: {
 					name: 1,
 				},
 			});
-
-			// Assert
 			await expect(endpoint.isChainNameAvailable(context)).rejects.toThrow(
 				'Chain name must be a string.',
 			);
 		});
 
-		it('should throw error if name is invalid format', async () => {
-			// Arrange
+		it('should throw error if name has invalid chars', async () => {
 			const context = createTransientModuleEndpointContext({
 				params: {
-					name: '@*#$*%&@((%$#@((',
+					name: '@*#(',
 				},
 			});
-
-			// Assert
 			await expect(endpoint.isChainNameAvailable(context)).rejects.toThrow(
 				`Invalid name property. It should contain only characters from the set [a-z0-9!@$&_.].`,
 			);
 		});
 
+		it(`should throw error if name length exceeds ${MAX_CHAIN_NAME_LENGTH}`, async () => {
+			const context = createTransientModuleEndpointContext({
+				params: {
+					name: 'a'.repeat(MAX_CHAIN_NAME_LENGTH + 1),
+				},
+			});
+			await expect(endpoint.isChainNameAvailable(context)).rejects.toThrow(
+				`Invalid name property. Length should be >= ${MIN_CHAIN_NAME_LENGTH} and <= ${MAX_CHAIN_NAME_LENGTH}.`,
+			);
+		});
+
+		it(`should not throw error if name length equals ${MIN_CHAIN_NAME_LENGTH}`, async () => {
+			const context = createTransientModuleEndpointContext({
+				params: {
+					name: 'a',
+				},
+			});
+			await expect(endpoint.isChainNameAvailable(context)).resolves.toStrictEqual({
+				result: false,
+			});
+		});
+
+		it(`should not throw error if name length equals ${MAX_CHAIN_NAME_LENGTH}`, async () => {
+			const context = createTransientModuleEndpointContext({
+				params: {
+					name: 'a'.repeat(MAX_CHAIN_NAME_LENGTH),
+				},
+			});
+			await expect(endpoint.isChainNameAvailable(context)).resolves.toStrictEqual({
+				result: false,
```

Comment:
Although the purpose of this test is to see if this function throws, the "expected result" for this test should be a `true` since the name hasn't need been registered and the name complies standard. I would suggest adding 
```
jest.spyOn(registeredNamesStore, 'has').mockResolvedValue(false);
```

to make the result true